repondeur/zam_repondeur/services/fetch/an/amendements.py

```diff
@@ -264,55 +232,113 @@ def _create_or_update_amendement(
-            parent=parent,
-            position=position,
-            id_discussion_commune=id_discussion_commune,
-            id_identique=id_identique,
-            matricule=matricule,
-            groupe=groupe,
-            auteur=auteur,
-            mission_titre=mission_titre,
-            mission_titre_court=mission_titre_court,
+        amendement = lecture.find_amendement(num)
+
+        action: Optional[Action] = None
+
+        if amendement is None:
+            action = CreateAmendement(
+                subdiv=subdiv,
+                parent_num_raw=parent_num_raw,
+                num=num,
+                rectif=rectif,
+                position=position,
+                id_discussion_commune=id_discussion_commune,
+                id_identique=id_identique,
+                matricule=matricule,
+                groupe=groupe,
+                auteur=auteur,
+                mission_titre=mission_titre,
+                mission_titre_court=mission_titre_court,
+                corps=corps,
+                expose=expose,
+                sort=sort,
+            )
+            return amendement, action
+
+        old_parent_num_raw = str(amendement.parent.num) if amendement.parent else ""
+
+        modified = any(
+            [
+                (amendement.article is None or subdiv != amendement.article.subdiv),
+                parent_num_raw != old_parent_num_raw,
+                rectif != amendement.rectif,
+                corps != amendement.corps,
+                expose != amendement.expose,
+                sort != amendement.sort,
+                id_discussion_commune != amendement.id_discussion_commune,
+                id_identique != amendement.id_identique,
+                matricule != amendement.matricule,
+                groupe != amendement.groupe,
+                auteur != amendement.auteur,
+                mission_titre != amendement.mission_titre,
+                mission_titre_court != amendement.mission_titre_court,
+                corps != amendement.corps,
+                expose != amendement.expose,
+                sort != amendement.sort,
+            ]
+        )
+        if modified:
+            action = UpdateAmendement(
+                amendement_num=amendement.num,
+                subdiv=subdiv,
+                parent_num_raw=parent_num_raw,
+                rectif=rectif,
+                position=position,
+                id_discussion_commune=id_discussion_commune,
+                id_identique=id_identique,
+                matricule=matricule,
+                groupe=groupe,
+                auteur=auteur,
+                mission_titre=mission_titre,
+                mission_titre_court=mission_titre_court,
+                corps=corps,
+                expose=expose,
+                sort=sort,
+            )
+        return amendement, action
+
+    def _set_fetch_progress(self, lecture: Lecture, position: int, total: int) -> None:
+        total = total + MAX_404
```

Comment:
I guess we should change the line `lecture.set_fetch_progress(1, len(lecture.amendements) * 2)` too in lectures views.